File: react/src/lib/services/OrganizationService.ts
import Organization from '../domain/entities/Organization';
import OrganizationRepository from '../domain/repositories/OrganizationRepository';

export default class OrganizationService {
    constructor(private organizationRepository: OrganizationRepository) {
    }

    async add(organization: Organization): Promise<Organization> {
        return this.organizationRepository.add(organization);
    }

    async update(organization: Organization): Promise<void> {
        return this.organizationRepository.update(organization);
    }

    async delete(id: string): Promise<void> {
        return this.organizationRepository.delete(id);
    }

    async findAll(): Promise<Organization[]> {
        return this.organizationRepository.findAll();
    }

    async findById(id: string): Promise<Organization> {
        return this.organizationRepository.findById(id);
    }
}

File: backend/src/main/java/com/equisoft/dca/backend/user/service/OrganizationService.java
package com.equisoft.dca.backend.user.service;

import java.util.List;
import java.util.Optional;

import javax.validation.Valid;

import com.equisoft.dca.backend.user.model.Organization;

public interface OrganizationService {

	Optional<Organization> findById(int id);

	Optional<Organization> findByName(String name);

	Organization save(@Valid Organization organization);

	Organization update(@Valid Organization organization);

	void deleteById(Integer id);

	List<Organization> findAll();
}
